### ISSUE TITLE
Eslint doesn't fail when calling undefined class method

### ISSUE THREAD

**Issue by mikefisher84** [eslint/eslint#8650]
I'm trying to configure eslint on my project and I'm noticing something that seems like a major issue. 

When calling a method in a class with this eslint always says it's ok even if it doesn't exist. For example. 

```
class Pizza {
	constructor() {
		
	}

	makePizza() {
           this.makeBagel();
	
	}
}
```

there is no method `makeBagel` but because its called as a method on `this` eslint doesn't flag it. Is there a way to prevent this as our application very heavily uses es6 classes in this manner. 

**Tell us about your environment**

* **ESLint Version:** `3.17.1`
* **babel-eslint Version:** `7.1.1`
* **Node Version:** `6.10.2`
* **yarn Version:**  `0.21.3`
* **Parser** `babel-eslint`

**Comment by ilyavolodin** [comment]
As far as I know, we don't have a rule like that. You can definitely create a custom rule that would check for properties of `this`. However, I don't think we would be able to add it to the core, due to the fact, that this rule becomes very useful when you use inheritance, but ESLint currently lints only one file at a time, so we wouldn't be able to check for existence of the properties on the base classes.

**Comment by not-an-aardvark** [comment]
Closing because the question has been answered.
